addons/payment_asiapay/models/payment_acquirer.py

```diff
@@ -0,0 +1,47 @@
+from odoo import fields, models
+
+from .const import CURRENCY_MAPPING
+
+
+class PaymentAcquirer(models.Model):
+    _inherit = "payment.acquirer"
+
+    def _domain_asiapay_currency_id(self):
+        xmlids = ["base.{}".format(key) for key in CURRENCY_MAPPING]
+        return [('id', 'in', [self.env.ref(xmlid).id for xmlid in xmlids])]
+
+    provider = fields.Selection(selection_add=[("asiapay", "Asiapay")], ondelete={"asiapay": "set default"})
+    asiapay_merchant_id = fields.Char(
+        string="AsiaPay Merchant Partner ID",
+        required_if_provider="asiapay",
+        help="The Merchant Partner ID is used to connect to Asiapay merchant account.",
+    )
+    asiapay_currency_id = fields.Many2one(
+        "res.currency",
+        required_if_provider="asiapay",
+        string="AsiaPay Currency Code",
+        domain=_domain_asiapay_currency_id,
+    )
+    asiapay_secure_hash = fields.Char(
+        string="AsiaPay Secure Hash",
+        help="The Secure Hash is used to ensure communications coming from Asiapay are valid and secured.",
+    )
+    asiapay_secure_hash_function = fields.Selection(
+        [('sha1', 'Sha1'), ('sha256', 'Sha256')],
+        required=True,
+        default='sha1',
```

Comment:
Is the default really `SHA1` on AsiaPay side? It's weird for them to offer SHA256 and not make it the default.

Reply:
It is weird but in fact we dont have control on the hashing method. This is set in Asiapay however it is not set by the merchant. It is set by Asiapay side.
![image](https://user-images.githubusercontent.com/85471608/182194401-25a4dc7b-5a4c-410c-a01b-1224a817f203.png)

Reply:
Ok, that could make sense if only AsiaPay enabled a SHA2 method for new accounts :smile: 

Let's keep the field in case it's not SHA1 for everyone but hide it in debug mode so that users don't have to figure that one out.